docs/API-functions.md

```diff
@@ -459,6 +502,126 @@ setting the `SRTO_RENDEZVOUS` option to true, and doing `srt_connect`.
+Socket group management
+-----------------------
+
+Group types are collected in an `SRT_GROUP_TYPE` type and they are currently:
+
+* `SRT_GTYPE_BROADCAST`: broadcast type, all links are actively used at once
+* `SRT_GTYPE_BACKUP`: backup type, idle links take over connection on disturbance
+* `SRT_GTYPE_BALANCING`: balancing type, share bandwidth usage between links
+
+Functions to be used on groups:
+
+### srt_create_group
+
+```
+SRTSOCKET srt_create_group(SRT_GROUP_TYPE type);
+```
+
+Creates a new group of type `type`. This is typically called on the
+caller side to be next used for connecting to the listener peer side.
+The group ID is of the same domain as socket ID, with the exception that
+the `SRTGROUP_MASK` bit is set on it, unlike for socket ID.
+
+### srt_include
+
+```
+int srt_include(SRTSOCKET socket, SRTSOCKET group);
+```
+
+This function adds a socket to a group. This is only allowed for unmanaged
+groups. No such group type is currently implemented.
+
+### srt_exclude
+
+```
+int srt_exclude(SRTSOCKET socket);
+```
+This function removes a socket from a group to which it currently belongs.
+This is only allowed for unmanaged groups. No such group type is currently
+implemented.
+
+### srt_groupof
+
+```
+SRTSOCKET srt_groupof(SRTSOCKET socket);
+```
+
+Returns the group ID of the socket, or `SRT_INVALID_SOCK` if the socket
+doesn't exist or it's not a member of any group.
+
+### srt_group_data 
+
+```
+int srt_group_data(SRTSOCKET socketgroup, SRT_SOCKGROUPDATA* output, size_t* inoutlen);
+```
+
+* `socketgroup` an existing socket group ID
+* `output` points to an output array
+* `inoutlen` points to a variable set to array's size
+
+This function obtains the current member state of the group specified in
+`socketgroup`. The `output` should point to the array of enough size to get
+all the elements and `inoutlen` to a variable preset to the size of this array.
+The current number of members will be written back to `inoutlen`. If the size
+is enough for the current number of members, the `output` array will be
+filled with group data and the function will return 0. Otherwise the array
+will not be filled and `SRT_ERROR` will be returned.
+
+- Returns:
+
+   * 0, if successful
+   * -1, on error
+
+- Errors:
+
+   * `SRT_EINVPARAM` reported if `socketgroup` is not an existing group ID
+   * `SRT_SUCCESS` if the array was too small
+
+
+
+### srt_connect_group
+
+```
+int srt_connect_group(SRTSOCKET group,
+                      const struct sockaddr* source /*nullable*/, int sourcelen,
+                      SRT_SOCKGROUPDATA name [], int arraysize);
+```
+
+This function does almost the same as calling `srt_connect` (or
+`srt_connect_bind`, if `source` is not NULL) in a loop for every item specified
+in `name` array. However if you did this in a blocking mode, the first call
+to `srt_connect` would block until the connection is established, whereas this
+function blocks until any of the specified connections is established. 
```

Comment:
This function does almost the same as calling `srt_connect` (or
`srt_connect_bind`, if `source` is not NULL) in a loop for every item specified
in `name` array. However, if you did this in blocking mode, the first call
to `srt_connect` would block until the connection is established, whereas this
function blocks until any of the specified connections is established.

Reply:
Not sure, but I can only see removed "a" in "did this in a blocking mode", am I seeing it right?